main/file.c

```diff
@@ -1288,22 +1290,41 @@ int ast_file_read_dirs(const char *dir_name, ast_file_on_file on_file, void *obj
 	return res;
 }
 
-int ast_streamfile(struct ast_channel *chan, const char *filename, const char *preflang)
+int ast_streamfile(struct ast_channel *chan, const char *filename,
+	const char *preflang)
 {
-	struct ast_filestream *fs;
-	struct ast_filestream *vfs=NULL;
+	struct ast_filestream *fs = NULL;
+	struct ast_filestream *vfs = NULL;
 	off_t pos;
 	int seekattempt;
 	int res;
+	char custom_filename[256];
+	char *tmp_filename;
+
+	/* If file with the same name exists in /var/lib/asterisk/sounds/custom directory, use that file.
+	 * Otherwise, use the original file*/
+
+	if (ast_opt_sounds_search_custom && !is_absolute_path(filename)) {
+		memset(custom_filename, 0, sizeof(custom_filename));
+		snprintf(custom_filename, sizeof(custom_filename), "custom/%s", filename);
+		fs = ast_openstream(chan, custom_filename, preflang);
+		if (fs) {
+			tmp_filename = custom_filename;
+			ast_verb(3, "Found file %s in custom directory\n", filename);
```

Comment:
This should definitely not be a verbose message, it should be a debug at least level 2 or 3.